AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Sync/InitialSync/InitialSyncOrchestrator.swift

```diff
@@ -164,7 +164,16 @@ final class AWSInitialSyncOrchestrator: InitialSyncOrchestrator {
             allMessages.joined(separator: "\n"),
             underlyingError
         )
-        return .failure(syncError)
+        
+        // send success if atleast one model succeeded
+        let syncableModelsCount = ModelRegistry.modelSchemas.filter { $0.isSyncable }.count
+        if syncableModelsCount == syncErrors.count {
+            return .failure(syncError)
```

Comment:
This seems like a reasonable change but thinking about this more, and looking at the start of this method, appears that this change will excessively return success. Let's think about the cascading impact of this change. Before we would fail fast, and DataStore will "re-sync" by attemping the entire sync lifecycle on the next DataStore operation. By not failing fast, eventual consistency will be much longer. 

Instead of this change, is it because we are getting an unauthorized error but not checking it properly here?
```
if syncErrors.isEmpty || syncErrors.allSatisfy(isUnauthorizedError) {
            return .successfulVoid
        }
```

In the example https://github.com/aws-amplify/amplify-swift/issues/3136 what is the sync error instance, why doesn't it return success at line 152?

Reply:
The bug happens when there is no user signed in and a `Guest` instance is saved locally. 
Now, there are two syncable models : `Guest` and `User`.
`User` gets `onReceiveCompletion()` with error result with `Unauthorized error` and append result to `syncErrors` array.
`Guest` gets `onReceiveCompletion()` with success.

When `makeCompletionResult()` is called - it returns `.failure()` and the `Guest` instance doesn't sync at all.

Reply:
why does `makeCompletionResult()` return `.failure()` if the syncing User contributes to one syncError which should satisfy `syncErrors.allSatisfy(isUnauthorizedError)` ?

Reply:
thanks for mentioning that. i have updated the logic for checking unauthorized error.